backends/fs/chroot/chroot-fs.cpp

```diff
@@ -0,0 +1,99 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "chroot-fs.h"
+
+ChRootFilesystemNode::ChRootFilesystemNode(const Common::String &root, POSIXFilesystemNode *node) {
+	_root = Common::normalizePath(root, '/');
+	_realNode = node;
+}
+
+ChRootFilesystemNode::ChRootFilesystemNode(const Common::String &root, const Common::String &path) {
+	_root = Common::normalizePath(root, '/');
+	_realNode = new POSIXFilesystemNode(root.stringByAppendingPathComponent(path));
+}
+
+ChRootFilesystemNode::~ChRootFilesystemNode() {
+	delete _realNode;
+}
+
+bool ChRootFilesystemNode::exists() const {
+	return _realNode->exists();
+}
+
+Common::String ChRootFilesystemNode::getDisplayName() const {
+	return getName();
+}
+
+Common::String ChRootFilesystemNode::getName() const {
+	return _realNode->AbstractFSNode::getDisplayName();
+}
+
+Common::String ChRootFilesystemNode::getPath() const {
+	Common::String path = _realNode->getPath();
+	if (path.size() > _root.size()) {
+		return Common::String(path.c_str() + _root.size());
+	}
+	return Common::String("/");
+}
+
+bool ChRootFilesystemNode::isDirectory() const {
+	return _realNode->isDirectory();
+}
+
+bool ChRootFilesystemNode::isReadable() const {
+	return _realNode->isReadable();
+}
+
+bool ChRootFilesystemNode::isWritable() const {
+	return _realNode->isWritable();
+}
+
+AbstractFSNode *ChRootFilesystemNode::getChild(const Common::String &n) const {
+	return new ChRootFilesystemNode(_root, (POSIXFilesystemNode *) _realNode->getChild(n));
```

Comment:
Formatting, we don't use a space between `(POSIXFilesystemNode *)` and `_realNode->getChild(n)`.